source/delegate/migrations/1_initial_migration.js

```diff
@@ -1,12 +1,16 @@
+const DelegateFactory = artifacts.require('DelegateFactory')
 const Delegate = artifacts.require('Delegate')
 
-module.exports = (deployer, network) => {
-  if (network == 'rinkeby' || network == 'mainnet') {
+module.exports = async(deployer, network) => {
     // fill in the addresses below
     const SWAP_ADDRESS = ''
     const INDEXER_ADDRESS = ''
     const OWNER_ADDRESS = ''
     const TRADE_WALLET_ADDRESS = ''
-    deployer.deploy(Delegate, SWAP_ADDRESS, INDEXER_ADDRESS, OWNER_ADDRESS, TRADE_WALLET_ADDRESS)
+    const FACTORY_PROTOCOL = ''
+
+  if (network == 'mainnet' || network == 'rinkeby') {
+    await deployer.deploy(DelegateFactory, SWAP_ADDRESS, INDEXER_ADDRESS, FACTORY_PROTOCOL)
+    await deployer.deploy(Delegate, SWAP_ADDRESS, INDEXER_ADDRESS, OWNER_ADDRESS, TRADE_WALLET_ADDRESS, FACTORY_PROTOCOL)
```

Comment:
Why are we deploying a delegate? Isn't that what the delegate factory is for? I feel like it'd be better to deploy a factory and then create a delegate from that

Reply:
Tbf I see that we dont ever really use this file

Reply:
As I understood it, it was difficult to verify the delegate from the delegate factory and so deploying a dummy one to verify was the only way to do it.

Reply:
This was custom deployed to allow the reuse of an existing indexer as opposed to redeploying new contracts.